src/sentry/static/sentry/app/views/organizationDiscover/discover.jsx

```diff
@@ -340,51 +375,18 @@ export default class OrganizationDiscover extends React.Component {
             )}
             {!shouldDisplayResult && (
               <React.Fragment>
-                <Box mt={1} mb={2}>
-                  <Heading>{t('Discover')}</Heading>
-                </Box>
+                <div>
```

Comment:
Can we drop this extra div that snuck in here?

Reply:
This div is actually the key to fixing the header shift bug. It blocks the flex layout from altering the `HeadingContainer` height, which would cause unexpected changes, even with identical HTML.

Why the height of this element changes without the wrapping div is pretty mysterious—I'm not sure if was switching from intrinsic to extrinsic height, or if the element was just growing or shrinking somehow based on the size of adjacent siblings. Sometimes flex is so 😵💀

Reply:
Here are two gifs that shows the difference with and without the div:

![wiggle](https://user-images.githubusercontent.com/435981/48864585-e4ccef00-ed81-11e8-85c9-2a62327e0b01.gif)

![wiggle-fixed](https://user-images.githubusercontent.com/435981/48864583-e4ccef00-ed81-11e8-99d1-2be6dcba8d67.gif)



![image](https://user-images.githubusercontent.com/435981/48864630-029a5400-ed82-11e8-95d5-8624410aeb98.png)

![image](https://user-images.githubusercontent.com/435981/48864667-1fcf2280-ed82-11e8-97af-e3134f7ae8b2.png)

Reply:
Looking at it again, I think it has to do with the way `flex` removes line-height from the text. Probably for centering purposes.

Reply:
Oh I see, that is subtle. I guess the other alternative is to put the height on the heading container then? The upside would be removing unnecessary dom node, downside would be hardcoding 22px in another line of css so i'm not sure if that's an improvement

Reply:
Hardcoding the height would cause bugs if the title wrapped, and we can't do `line-height: 0` either unfortunately, because if the title wrapped the lines would be on top of each other.

Reply:
And yeah it's pretty subtle lol. I promise it's a bit more obvious IRL! I'm admittedly not the most detail-oriented person in the world and I noticed.

Reply:
Yeah, that's fair. TBH my main concern is just that it's extremely easy to forget why these workarounds are in place later and break this (so if there's anyway to solve it via css it would be better). If not then 👍

Reply:
Yeah I agree with that philosophy. I'm just not sure how to make it happen here. We could be deliberate and apply `display: block` to the element with a comment.